test/test.js

```diff
@@ -0,0 +1,259 @@
+/*
+* @Author: alutun
+* @Date:   2017-07-03 14:01:38
+* @Last Modified by:   alutun
+* @Last Modified time: 2017-07-03 16:19:16
+*/
+
+var supertest = require("supertest");
+var should = require("should");
+var dateFormat = require('dateformat');
+var now = new Date();
+var util = require('util');
+
+var server = supertest.agent("http://localhost:3000");
+
+var savedId;
+// UNIT test begin
+
+describe("Basic test",function(){
+  it("should return home page",function(done){
+    // Test home page API
+    server
+    .get("/")
+    .expect("Content-type",/json/)
+    .expect(200) // THis is HTTP response
+    .end(function(err,res){
+      // HTTP status should be 200
+      res.status.should.equal(200);
+      done();
+    });
+  });
+
+  it("Should return 404 for wrong endpoint", function(done){
+  	server
+  	.get("/random")
+    .expect(404)
+    .end(function(err,res){
+      res.status.should.equal(404);
+      done();
+  	});
+   });
+});
+
+describe("PUT /user",function(){
+
+	it("Simple Put Test", function(done){
+		server
+		.post("/user")
+	 	.send({name: "toto", dob: dateFormat(now, "isoDateTime"), address: "Shanghai", description: "This user is a test"})
+		.expect("Content-type",/json/)
+	    .expect(200)
+	    .end(function(err,res){
+	      res.status.should.equal(200);
+	      // console.log("res body is : " + util.inspect(res.body));
+	      res.body.name.should.equal("toto");
+	      res.body.description.should.equal("This user is a test");
+	      res.body.address.should.equal("Shanghai");
+	      res.body.should.have.property("id").which.is.a.Number();
+	      savedId = res.body.id;
```

Comment:
So if I want to make each test independent from each other I will do as follow. 

For example for DELETE endpoint Nested in the test 

1. I would first create an user json object which I will use to create a user with PUT.
2. Then Once the PUT is successful I will update my user object with the ID sent back inside my PUT answer. 
3. Then using that user json object I would do my test on the DELETE endpoint. 

So basically within each test, I would create the necessary HTTP resources needed to perform them independently. 

Let me know If my answer is clear enough.